app/models/notification.rb

```diff
@@ -41,7 +41,11 @@ def email_the_user(target, actor)
   end
 
   def set_read_state( read_state )
-    self.update_attributes( :unread => !read_state )
+    if self.new_record?
```

Comment:
We shouldn't change the code only to make tests running. Please change the tests that they use saved notifications, so the tests correspond to the real conditions.